src/kareem/core.clj

```diff
@@ -232,12 +246,92 @@
       (respond-to-messages! msg-events))
   (response {}))
 
+;; ------------------
+;; sms
+
+(defn parse-int [s]
+  (Long. (re-find #"\d+" s)))
+
+(defn xml-response [body]
+  {:status 200
+   :headers {"content-type" "application/xml"}
+   :body (str "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" body)})
+
+(defn text-twiml [message]
+  (format "<Response>
+            <Message>
+              <Body>
+              %s
+              </Body>
+            </Message>
+          </Response>"
+          message))
+
+(defn ->atts [params]
+  (let [num-media (parse-int (:NumMedia params))]
+    (->> num-media
+         range
+         (keep (fn [i]
+                 (let [content-type-k (keyword (str "MediaContentType" i))
+                       url-k (keyword (str "MediaUrl" i))
+                       content-type (content-type-k params)]
+                   ;; TODO(stopachka)
+                   ;; remove "type", and use content-type in the ui
+                   ;; this will allow us to be more specific about extensions
+                   (when (= content-type "image/jpeg")
+                     {:type "image"
+                      :content-type content-type
+                      :payload {:url (url-k params)}})))))))
+
+(defn ->message [params]
+  (let [text (:Body params)
+        from (:From params)
+        atts (->atts params)]
+    {:sender {:id (parse-int from)
+              :from from}
+     :timestamp (System/currentTimeMillis)
+     :message {:text text
+               :attachments atts}}))
+
+(defn parse-intent [{:keys [text attachments]}]
+  (let [text (-> text str string/lower-case string/trim)]
+    (cond
+      (seq attachments)
+      ::log
+
+      (string/includes? text "history")
+      ::history
+
+      :else ::log)))
+
+(defn post-sms [{:keys [params]}]
+  (let [text-res #(xml-response (text-twiml %))
+        {:keys [sender message] :as evt} (->message params)
+        intent (parse-intent message)]
+    (condp = intent
```

Comment:
super nit -- could use `case` instead here, right? https://puredanger.github.io/tech.puredanger.com/2010/05/30/learning-clojure-15-case/ 🤷‍♂️

Reply:
love it!!